### ISSUE TITLE
`since` refactor

### ISSUE THREAD

**Issue by guidopetri** [guidopetri/chess-pipeline#5]
The way that the `since` parameter works right now is kind of clunky: it'll run `FetchLichessApiPGN` and request from that date on. This is crappy when you miss a day or several (due to bugs...), since then you have to find the right date to request from, or delete all rows from the `table_updates` table.

Refactoring `since` to be a parameter in `CopyGames` should allow us to dynamically call the API endpoint in a better fashion. There could be other ways of structuring the Tasks in order to achieve a better way of doing this, too - maybe not have it be dynamic, but rather only one day at a time, which would let us use luigi staples like `RangeDaily` easier.

**Comment by guidopetri** [comment]
I've written some code for this in a separate branch, but I'd like to test it before merging to master. So I'm waiting on #20 to finish before making any further progress here.

**Comment by guidopetri** [comment]
Currently in testing on the `dev` branch.

**Comment by guidopetri** [comment]
Many small problems here... the filenames can't get overwritten (dates must be in the temp files), a parentheses missing at the end of the `chess_pipeline.py` file (dunno how I missed that one), `until` not working as supposed to in `CopyGames` (requiring me to re-define `requires(self)`), `dl_day` defaults missing...

These are all small, but easy fixes (done in 0046ead ). However, now that I'm thinking about it - maybe there should be a function/task to collect all the input and put it into a single transaction? Would this even work in the spirit of this refactor? I imagine it would have to be in the `TransactionFactTable` Task...

**Comment by guidopetri** [comment]
After thinking about it, I don't think putting everything in a single transaction is any good (unless if I were to go into the Luigi internals and modify what gets put in `table_updates` manually.... which I'm not willing to do). So for now, this'll go back to testing after fixing the filenames.

Unfortunately, it seems like the dynamic job requirements aren't working properly. I'm investigating this. On the other hand, it is certainly issuing the jobs correctly the first time it runs. Here's an example scenario:

- Run CopyGames with `until` set to yesterday (the default). This creates the jobs correctly.
- Run CopyGames right after with `until` set to April 1. This says the job is already done.

I'm not really sure what's causing this.

**Comment by guidopetri** [comment]
Examining an example run... it does seem like this way takes a **LOT** longer to process. Anecdotally (I didn't measure), it seems like all the postgres copy functions are taking a long time. I assume this is because of the giant overhead that I now have, compared to earlier when it was just one single transaction...

This begs the question: is this even worth doing?

**Comment by guidopetri** [comment]
After thinking about it for a long time, I've decided to not continue with this refactor. The substantial performance decrease in copying data over to Postgres is just not worth it. This can maybe be revisited if:

- I decide to use a "streaming" database (I think that's what they're called?) like Redis, which can then copy to a regular database later maybe?
- I start collecting data on several users at once and copying them all pooled together (which would make it so that I have lots of data copying at once to Postgres)

Since right now I'm only looking at 2-3 users, I won't bother. The code for this will remain in the `since_refactor` branch in case I want to take it up again later.
